src/Data/Bin/Ordering.agda

```diff
@@ -0,0 +1,790 @@
+suc[x]≤double[x] x =  x<⇒suc[x]≤ {x} {double x} ∘ x<double[x] x
+
+x≤suc[y]*x :  ∀ (x y) → x ≤ (suc y) * x
+x≤suc[y]*x x y =  begin
+  x             ≤⟨ x≤x+y x (y * x) ⟩
+  x + y * x     ≈⟨ sym (suc-* y x) ⟩
+  (suc y) * x   ∎
+  where
+  open IneqReasoning
+
+x≤double[x] :  ∀ x → x ≤ double x
+x≤double[x] x =  begin
+  x         ≤⟨ x≤x+y x x ⟩
+  x + x     ≈⟨ sym (double[x]≡x+x x) ⟩
+  double x  ∎
+  where
+  open IneqReasoning
+
+suc[x]<2[1+x] :  ∀ x → suc x < 2[1+ x ]
+suc[x]<2[1+x] x =  begin-strict
+  suc x           <⟨ x<double[x] (suc x) suc≢0  ⟩
+  double (suc x)  ≈⟨ sym (2[1+-as∘ x) ⟩
+  2[1+ x ]        ∎
+  where
+  open IneqReasoning
+
+x<2[1+x] :  ∀ x → x < 2[1+ x ]
+x<2[1+x] x =  <-trans {x} {suc x} {2[1+ x ]} (x<suc[x] x) (suc[x]<2[1+x] x)
+
+x<1+[2x] :  ∀ x → x < 1+[2 x ]
+x<1+[2x] x =  begin-strict
+  x               <⟨ x<suc[x] x ⟩
+  suc x           ≤⟨ suc-mono-≤ {x} {double x} (x≤double[x] x) ⟩
+  suc (double x)  ≈⟨ sym (1+[2-as∘ x) ⟩
+  1+[2 x ]        ∎
+  where
+  open IneqReasoning
+
+double[x]<1+[2x] :  ∀ x → double x < 1+[2 x ]
+double[x]<1+[2x] x =  begin-strict
+  double x     <⟨ x<suc[x] (double x) ⟩
+  suc (double x)   ≈⟨ sym (1+[2-as∘ x) ⟩
+  1+[2 x ]     ∎
+  where
+  open IneqReasoning
+
+pred-mono-≤ :  pred Preserves _≤_ ⟶ _≤_
+pred-mono-≤ {x} {y} x≤y =  begin
+  pred x             ≈⟨ cong pred (sym (fromℕ-toℕ x)) ⟩
+  pred (fromℕ m)     ≈⟨ sym (fromℕ-pred m) ⟩
+  fromℕ (ℕ.pred m)   ≤⟨ fromℕ-mono-≤ (ℕₚ.pred-mono m≤n) ⟩
+  fromℕ (ℕ.pred n)   ≈⟨ fromℕ-pred n ⟩
+  pred (fromℕ n)     ≈⟨ cong pred (fromℕ-toℕ y) ⟩
+  pred y             ∎
+  where
+  open IneqReasoning
+  m = toℕ x;  n = toℕ y;  m≤n = toℕ-mono-≤ x≤y
+
+x<1⇒x≡0 :  ∀ {x} → x < 1B → x ≡ zero
+x<1⇒x≡0 {x} x<1 =  ≤0⇒≡ x≤0
+  where
+  open IneqReasoning
+  suc[x]≤1 = x<⇒suc[x]≤ {x} {1B} x<1
+  x≤0 = begin
+    x             ≈⟨ sym (pred-suc x) ⟩
+    pred (suc x)  ≤⟨ pred-mono-≤ {suc x} {1B} suc[x]≤1 ⟩
+    pred 1B       ≈⟨ refl ⟩
+    zero          ∎
+
+x≢0-nor-1⇒1<x :  ∀ {x} → x ≢ zero → x ≢ 1B → 1B < x
+x≢0-nor-1⇒1<x {x} x≢0 x≢1  with <-cmp 1B x
+... | tri< 1<x _   _   =  1<x
+... | tri≈ _   1≡x _   =  contradiction (sym 1≡x) x≢1
+... | tri> _   _   1>x =  contradiction (x<1⇒x≡0 1>x) x≢0
+
+_<ₙon_ :  Rel Bin 0ℓ
+_<ₙon_ =  ℕ._<_ on toℕ
+
+<⇒<ₙon :  ∀ {x y} → x < y → x <ₙon y
+<⇒<ₙon x<y =  toℕ-mono-< x<y
+
+<ₙon⇒< :  ∀ {x y} → x <ₙon y → x < y
+<ₙon⇒< {x} {y} xN<yN =  subst₂ _<_ (fromℕ-toℕ x) (fromℕ-toℕ y) xN'<yN'
+  where
+  xN'<yN' = fromℕ-mono-< xN<yN
+
+------------------------------------------------------------------------------
+-- StrictTotalOrder and DecTotalOrder instances for Bin.
+------------------------------------------------------------------------------
+
+<-isStrictTotalOrder : IsStrictTotalOrder _≡_ _<_
+<-isStrictTotalOrder = record
+  { isEquivalence =  isEquivalence
+  ; trans         =  λ {x y z} → <-trans {x} {y} {z}
+  ; compare       =  <-cmp
+  }
+
+<-strictTotalOrder : StrictTotalOrder _ _ _
+<-strictTotalOrder = record
```

Comment:
For packages like these, there's no need to provide anything except the `is` proof.

Reply:
Fixed everywhere.